power/src/tests.rs

```diff
@@ -129,3 +129,189 @@ fn increase_multiple_regression() {
+}
+
+#[test]
+fn two_way_anova_test() {
+    let k = "5";
+    let q = "10";
+    let join = with_rest("twoWayANOVA");
+    let extra = json!({"k": k, "q": q, "analysis": "alpha"});
+    test_interface(&join(&extra), 0.465);
+    let extra = json!({"k": k, "q": q, "analysis": "power"});
+    test_interface(&join(&extra), 0.560);
+    let extra = json!({"k": k, "q": q, "analysis": "es"});
+    test_interface(&join(&extra), 0.770);
+    let extra = json!({"k": k, "q": q, "analysis": "n"});
+    test_interface(&join(&extra), 107.0);
+}
+
+#[test]
+fn ancova_test() {
+    let k = "5";
+    let q = "10";
+    let p = "2";
+    let join = with_rest("ANCOVA");
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "alpha"});
+    test_interface(&join(&extra), 0.469); // G*Power gives 0.467
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "power"});
+    test_interface(&join(&extra), 0.553); // G*Power gives 0.555
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "es"});
+    test_interface(&join(&extra), 0.775); // G*Power gives 0.773
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "n"});
+    test_interface(&join(&extra), 107.0);
+
+    let k = "10";
+    let q = "50";
+    let p = "10";
+    let join = with_rest("ANCOVA");
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "alpha"});
+    test_interface(&join(&extra), 0.826); // G*Power gives 0.825
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "power"});
+    test_interface(&join(&extra), 0.156); // G*Power gives 0.158
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "es"});
+    test_interface(&join(&extra), 1.357); // G*Power gives 1.344
+    let extra = json!({"k": k, "q": q, "p": p, "analysis": "n"});
+    test_interface(&join(&extra), 204.0);
+}
+
+#[test]
+fn between_repeated_anova_test() {
+    let k = "2";
+    let m = "4";
+    let rho = "0.5";
+    let join = with_rest("betweenRepeatedANOVA");
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "alpha"});
+    test_interface(&join(&extra), 0.008);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "power"});
+    test_interface(&join(&extra), 0.992);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "es"});
+    test_interface(&join(&extra), 0.411);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "n"});
+    test_interface(&join(&extra), 36.0);
+
+    let k = "5";
+    let m = "10";
+    let rho = "0.75";
+    let join = with_rest("betweenRepeatedANOVA");
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "alpha"});
+    test_interface(&join(&extra), 0.125);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "power"});
+    test_interface(&join(&extra), 0.880);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "es"});
+    test_interface(&join(&extra), 0.566);
+    let extra = json!({"k": k, "m": m, "rho": rho, "analysis": "n"});
+    test_interface(&join(&extra), 65.0);
```

Comment:
Was there a specific reason to test multiple values for `k`, `m`, and `rho`?

Reply:
Just robustness. I want to see if different `rho`s would give the same G*Power result. Then I thought, well might as well test different stuff since we are already changing `rho`.